web/test/pages/onboarding/onboarding-starting.test.tsx

```diff
@@ -198,17 +197,55 @@ describe("onboarding - starting a business", () => {
           email: "email@example.com",
         },
       };
-      mockApi.postSelfReg.mockResolvedValue({
-        authRedirectURL: "",
-        userData: {
-          ...expectedUserData,
-          user: { ...expectedUserData.user, myNJUserKey: "12345" },
-        } as UserData,
+
+      mockApi.postNewsletter.mockImplementation((request) =>
+        Promise.resolve({
+          ...request,
+          user: {
+            ...request.user,
+            externalStatus: {
+              ...request.user.externalStatus,
+              newsletter: { status: "SUCCESS", success: true },
+            },
+          },
+        })
+      );
+
+      mockApi.postUserTesting.mockImplementation((request) =>
+        Promise.resolve({
+          ...request,
+          user: {
+            ...request.user,
+            externalStatus: {
+              ...request.user.externalStatus,
+              userTesting: { status: "SUCCESS", success: true },
+            },
+          },
+        })
+      );
+
+      expect(mockApi.postNewsletter).toHaveBeenCalledWith({
+        ...expectedUserData,
+        user: { ...expectedUserData.user, externalStatus: {} },
+      });
+
+      expect(mockApi.postUserTesting).toHaveBeenCalledWith({
+        ...expectedUserData,
+        user: {
+          ...expectedUserData.user,
+          externalStatus: { newsletter: { status: "SUCCESS", success: true } },
+        },
```

Comment:
again, this feels like it should be not part of this test. at the very least, it makes the test so long. If necessary, could we move the mocking step to the beforeEach or something?